File: Submissions/Hello_2019/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  string t;
  cin >> t;

  string answer = "NO";

  for (int i = 0; i < 5; i++) {
    string card;
    cin >> card;
    if (t[0] == card[0] || t[1] == card[1]) {
      answer = "YES";
      break;
    }
  }

  cout << answer << endl;

}

File: Submissions/Hello_2019/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    String T = sc.next();

    String [] t = T.split("");

    String answer = "NO";

    for (int i = 0; i < 5; i++) {
      String Card = sc.next();
      String [] card = Card.split("");
      if (t[0].equals(card[0]) || t[1].equals(card[1])) {
        answer = "YES";
        break;
      }
    }

    sc.close();

    System.out.println(answer);

  }

}